internal/services/conditionalaccess/conditional_access_policy_resource.go

```diff
@@ -172,18 +174,53 @@ func conditionalAccessPolicyResource() *schema.Resource {
 						},
 
 						"client_app_types": {
-							Type:     schema.TypeList,
-							Required: true,
+							Type:             schema.TypeList,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 								ValidateFunc: validation.StringInSlice([]string{
-									"all",
-									"browser",
-									"mobileAppsAndDesktopClients",
-									"exchangeActiveSync",
-									"easSupported",
-									"other",
-								}, false),
+									msgraph.ConditionalAccessClientAppTypeAll,
+									msgraph.ConditionalAccessClientAppTypeBrowser,
+									msgraph.ConditionalAccessClientAppTypeEasSupported,
+									msgraph.ConditionalAccessClientAppTypeExchangeActiveSync,
+									msgraph.ConditionalAccessClientAppTypeMobileAppsAndDesktopClients,
+									msgraph.ConditionalAccessClientAppTypeOther,
+								}, true),
+							},
+						},
+
+						"devices": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"device_filter": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"mode": {
+													Type:             schema.TypeString,
+													Required:         true,
+													DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
how come we are suppressing case and not normalizing it to what is expected/correct?

Reply:
Good point - the correct casing isn't always obvious as the docs are inconsistent, but I think you're right it makes sense to normalize these (and make judgement calls for consistency as necessary) 👍

Reply:
I'll do some more testing and see if we get any flapping diffs due to casing, if not I'll pull out the diff suppression